tests/kernel/sched/schedule_api/src/user_api.c

```diff
@@ -9,7 +9,7 @@
 struct k_thread user_thread;
 K_SEM_DEFINE(user_sem, 0, 1);
 
-ZTEST_BMEM volatile bool thread_was_preempt;
+ZTEST_BMEM volatile int thread_was_preempt;
```

Comment:
why did you change the type from `bool` to `int`?

Reply:
To make it a machine word, some architectures generate more code for bytewise reads, and being volatile it's going to be repeated every time.  But really there's no particularly strong reason to have it in the commit.